packages/odyssey-react/src/components/Button/Button.module.scss

```diff
@@ -44,24 +45,24 @@
   }
 
   &:disabled {
-    border-color: transparent;
+    border-color: var(--ColorBorderPrimaryDisabled);
```

Comment:
This is an interesting case. With Primary and Danger variants, Design had previously expected them to appear "borderless" regardless of theme. 

i think this is beneficial to the component and keeps the property application standardized across variants. However, we'll want to suss out any Okta-internal design constraints around theming. Ex: "Should Designers reserve bordered buttons for Secondary styles?"

Reply:
@edburyenegren-okta yes this is tricky, I agree strongly that consistency is a valuable part of the public interface we want to build. Unfortunately, the application of component theme variables **isn't** standardized across variants because of how our class based styles make use of the cascade. Each variant only overrides certain styles of the .root class for Button, which means for now, variants expose different sets of component theme variables. We have a leaky abstraction here where internal implementation details are leaking through to the public interface.

In the future we should refactor such that we are **starting** with the API we want to expose, trying to keep it consistent across variants as you mentioned, and changing our internal implementation to allow for exposing this better interface. Yes it will require perhaps more granular styles to be written, or even writing styles that aren't strictly necessary (without a theme being applied) given the way the cascade works but that is fine so long as its abstracted away behind a better public API.